app/src/main/java/org/mozilla/tv/firefox/focus/FocusRepo.kt

```diff
@@ -241,6 +244,25 @@ class FocusRepo(
                 nextFocusLeftId = nextFocusLeftId))
     }
 
+    private fun getTurboButtonFocusState(
+        focusedTurboButtonNode: FocusNode,
+        sessionState: SessionRepo.State
+    ): State {
+
+        assert(focusedTurboButtonNode.viewId == R.id.turboButton)
+
+        val nextFocusLeftId = when {
+            // TODO: this is a duplicating existing logic in the ToolbarVM, may fall out of sync
+            sessionState.currentUrl == URLs.APP_URL_HOME -> R.id.turboButton
+            else -> R.id.pinButton
+        }
```

Comment:
There's an edge case here where pin is disabled and forward is enabled.

-> Open app
-> Select any tile
-> Hit back button

Reply:
Ah, nice one

Reply:
@Baron-Severin hmm it looks like we can't go back to `about:home` after selecting a tile and according to `ToolbarViewModel`, `pinEnabled = !sessionState.currentUrl.isEqualToHomepage()`

Reply:
I'm not sure of the code path, but I'm able to reproduce the state by following those steps.

<img width="1268" alt="Screen Shot 2019-05-24 at 1 06 51 PM" src="https://user-images.githubusercontent.com/13170306/58353961-d5a82600-7e24-11e9-9ef1-03f61cdd5b66.png">

Reply:
Ah okay so you can go back from within WebRenderFragment but not in Overlay 🤔